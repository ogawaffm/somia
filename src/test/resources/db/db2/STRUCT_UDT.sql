DROP TABLE IF EXISTS TEST.STRUCT_UDT
/
BEGIN
    DECLARE CONTINUE HANDLER FOR SQLSTATE '42704' BEGIN
    END; -- Type does not exist
    EXECUTE IMMEDIATE 'DROP TYPE TEST.STRUCT_TYPE';
END
/
BEGIN
    DECLARE CONTINUE HANDLER FOR SQLSTATE '42704' BEGIN
    END; -- Type does not exist
    EXECUTE IMMEDIATE 'DROP TYPE TEST.SUB_STRUCT_TYPE';
END
/
BEGIN
    DECLARE CONTINUE HANDLER FOR SQLSTATE '42704' BEGIN
    END; -- Type does not exist
    EXECUTE IMMEDIATE 'DROP TYPE TEST.CHAR_10_TYPE';
END
/
CREATE TYPE TEST.CHAR_10_TYPE AS CHAR(10)
/
CREATE TYPE TEST.SUB_STRUCT_TYPE AS (
    -- BOOLEAN and LOBS are not supported as fields in a structured type
    SUB_INTEGER_COL INTEGER,
    SUB_CHAR_10_TYPE_COL TEST.CHAR_10_TYPE
    ) MODE DB2SQL
/
CREATE TYPE TEST.STRUCT_TYPE AS (
    INTEGER_COL INTEGER,
    TIMESTAMP_12_COL TIMESTAMP(12),
    CHAR_10_TYPE_COL TEST.CHAR_10_TYPE,
    SUB_STRUCT_TYPE_COL TEST.SUB_STRUCT_TYPE
    ) MODE DB2SQL
/
CREATE TABLE TEST.STRUCT_UDT
(
    INTEGER_COL      INTEGER,
    STRUCT_TYPE_COL  TEST.STRUCT_TYPE,
    CHAR_10_TYPE_COL TEST.CHAR_10_TYPE
)