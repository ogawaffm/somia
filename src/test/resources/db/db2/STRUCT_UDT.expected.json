{
  "name": "STRUCT_UDT",
  "catalogName": "",
  "schemaName": "TEST",
  "typeName": "TABLE",
  "comment": "",
  "columns": [
    {
      "name": "INTEGER_COL",
      "type": {
        "name": "INTEGER",
        "typeNumber": 4,
        "className": "java.lang.Integer",
        "precision": 10,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": true,
        "isCurrency": false,
        "columnDisplaySize": 11,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "STRUCT_TYPE_COL",
      "type": {
        "name": "\"TEST\".\"STRUCT_TYPE\"",
        "typeNumber": 2002,
        "className": "java.sql.Struct",
        "precision": 16,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHAR_10_TYPE_COL",
      "type": {
        "name": "\"TEST\".\"CHAR_10_TYPE\"",
        "typeNumber": 2001,
        "className": "java.lang.Object",
        "precision": 10,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    }
  ],
  "indexes": [
    {
      "name": "",
      "columns": [],
      "isUnique": false,
      "comment": "",
      "qualifierName": "",
      "isClustered": false,
      "isHashed": false,
      "isStatistic": true,
      "whereClause": "",
      "cardinality": 0,
      "pages": 0
    }
  ],
  "collation": "",
  "primaryKey": null,
  "bestRowIdentifier": null
}
